lib/scheduler.js

```diff
@@ -172,19 +173,31 @@ class Scheduler {
         }
 
         if (schedule.time) {
-            let start = schedule.time.start;
-            let times;
-            if (start === 'sunrise' || start === 'sunset') {
-                times = this.suncalc.getTimes(new Date(), this.latitude, this.longitude);
-                start = times[start];
+            let start = schedule.time.start,
+                end = schedule.time.end,
+                now = new this.Date(),
+                times,
+                astroNameStart = this._getAstroName(start),
+                astroNameEnd = this._getAstroName(end);
+
+            if (now.getDate() !== this.todaysAstroTimes['sunrise']) {
```

Comment:
Should this compare not go to this.todaysAstroTimes['sunrise'].getDate() ?? Then this compare wopuld make sense ... When "now" has an other day then current sunrise then update it because most likely "new day" ... right?

Reply:
for the yesterday stuff: 
* there is a schedule for `nadir` and `nadir` happens past midnight, let's say at 00:30am
* if you start your script today noon(4-22), you would expect the trigger at 00:30 tomorrow(4-23)
* since Scheduler recalcs every minute, calculations past midnight will bring nadir time for 4-24 which potentially could be 00:20, or 00:35.
* that's why shifting is needed. It doesnt matter that results calculated thruout the day are 'wrong' because of that, only calculation right before the trigger is used anyway
* this is different from astro schedule, where calculation is done only once and then a timeout is set. There, also a 'wrong' time is used for past midnight events to force recalculation at midnight.

for compare: 
* before, suncalc was invoked every minute(for "start":"astroevent" schedules). this was way too much because times would only change once a day
* now these values are stored in todaysAstroTimes and yesterdaysAstroTimes. It must be ensured that they contain valid times before using them. you can see it like invocation of suncalc was replaced by comparision. this._setAstroVars will only be invoked once a day, though.
* i had seen no other way to update these vars at midnight, maybe a timeout similar to the one used by Scheduler would do, but only for once a day at midnight
* comparision is done at that place because both following comparisions for `start` and `end` must rely on these vars

Reply:
understood

Reply:
But @winnyschuster ... for the question ... this should be 

this.todaysAstroTimes['sunrise'].getDate() 

or????

Reply:
arrrgh, how embarrassing is this? to be honest, i didn't get your question right. When i've read _go to_ i immediately stopped reading and answered like _**no, this statement must not go to a different place**_. owe you a beer or a test case, i'm afraid there is more to come middle/long term. in fact i was thinking about a test case for that yesterday thing but didn't know how to do, mainly because Scheduler cuts seconds and at this time nadir changes only a few seconds a day so you wouldn't see a difference in results. Anyway, i was hoping that pr is now getting accepted, thanks for your patience, i promise to do better next time(s) :-)